huaweicloud/services/rds/resource_huaweicloud_rds_backup.go

```diff
@@ -0,0 +1,489 @@
+	}
+	return bodyParams
+}
+
+func buildCreateBackupDatabasesChildBody(d *schema.ResourceData) []map[string]interface{} {
+	rawParams := d.Get("databases").([]interface{})
+	if len(rawParams) == 0 {
+		return nil
+	}
+
+	params := make([]map[string]interface{}, len(rawParams))
+	for i, v := range rawParams {
+		raw := v.(map[string]interface{})
+		params[i] = map[string]interface{}{
+			"name": utils.ValueIngoreEmpty(raw["name"]),
+		}
+	}
+
+	return params
+}
+
+func buildCreateBackupWaitingQueryParams(d *schema.ResourceData) string {
+	res := ""
+	res = fmt.Sprintf("%s&backup_id=%v", res, d.Id())
+
+	if v, ok := d.GetOk("instance_id"); ok {
+		res = fmt.Sprintf("%s&instance_id=%v", res, v)
+	}
+
+	if res != "" {
+		res = "?" + res[1:]
+	}
+	return res
+}
+
+func createBackupWaitingForStateCompleted(ctx context.Context, d *schema.ResourceData, meta interface{}, t time.Duration) error {
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"PENDING"},
+		Target:  []string{"COMPLETED"},
+		Refresh: func() (interface{}, string, error) {
+			config := meta.(*config.Config)
+			region := config.GetRegion(d)
+			// createBackupWaiting: missing operation notes
+			var (
+				createBackupWaitingHttpUrl = "v3/{project_id}/backups"
+				createBackupWaitingProduct = "rds"
+			)
+			createBackupWaitingClient, err := config.NewServiceClient(createBackupWaitingProduct, region)
+			if err != nil {
+				return nil, "ERROR", fmt.Errorf("error creating Backup Client: %s", err)
+			}
+
+			createBackupWaitingPath := createBackupWaitingClient.Endpoint + createBackupWaitingHttpUrl
+			createBackupWaitingPath = strings.Replace(createBackupWaitingPath, "{project_id}", createBackupWaitingClient.ProjectID, -1)
+
+			createBackupWaitingqueryParams := buildCreateBackupWaitingQueryParams(d)
+			createBackupWaitingPath = createBackupWaitingPath + createBackupWaitingqueryParams
+
+			createBackupWaitingOpt := golangsdk.RequestOpts{
+				KeepResponseBody: true,
+				OkCodes: []int{
+					200,
+				},
+			}
+			createBackupWaitingResp, err := createBackupWaitingClient.Request("GET", createBackupWaitingPath, &createBackupWaitingOpt)
+			if err != nil {
+				return nil, "ERROR", err
+			}
+
+			createBackupWaitingRespBody, err := utils.FlattenResponse(createBackupWaitingResp)
+			if err != nil {
+				return nil, "ERROR", err
+			}
+			status, err := jmespath.Search(`backups[0].status`, createBackupWaitingRespBody)
+			if err != nil {
+				return nil, "ERROR", fmt.Errorf("error parse %s from response body", `backups[0].status`)
+			}
+
+			if utils.StrSliceContains(strings.Split(`FAILED`, ","), status.(string)) {
+				return createBackupWaitingRespBody, status.(string), nil
+			}
+
+			if utils.StrSliceContains(strings.Split(`COMPLETED`, ","), status.(string)) {
+				return createBackupWaitingRespBody, "COMPLETED", nil
+			}
+
+			return createBackupWaitingRespBody, "PENDING", nil
+		},
+		Timeout:      t,
+		Delay:        5 * time.Second,
```

Comment:
is 5 seconds too short? suggest extending the delay time.

Reply:
ok, change to 30